miio/discovery.py

```diff
@@ -207,7 +207,7 @@ def check_and_create_device(self, info, addr) -> Optional[Device]:
                     _LOGGER.info(
                         "%s: token: %s",
                         v(info),
-                        pretty_token(dev.do_discover().checksum),
+                        pretty_token(dev.command_sender.do_discover().checksum),
```

Comment:
I think we should move `do_discover` out from the protocol to be a separate `handshake()` or `hello()` for the device. The first step could be to create a wrapper as with `send()` for this, though.